EIPS/eip-7702.md

```diff
@@ -23,63 +23,126 @@ There is a lot of interest in adding short-term functionality improvements to EO
 
-Hence, there is a very simple transformation from "existing EIP-3074 workflows" into workflows under this new scheme.
+#### In-protocol revocation
+
+A hotly debated element of this EIP is the need for in-protocol revocation. Although it is possible to implement revocation logic within delegated code, some say this isn't sufficient and it is the duty of the protocol to provide a revocation option of last resort.
+
+From a first-principles perspective, users should only delegate control of their account to code they trust. Trust is often by proxy because most users are unable to verify smart contract code for themselves. This framing is similar to trusting client software to behave correctly, wallet software to not steal keys, and hardware wallets to not have backdoors. So long as the components are open source and the organizations behind the software have some reputation at stake, this will be a perfectly safe environment for users.
+
+Writing safe and bug-free smart contracts is difficult. However, not all code paths are create equal. There are certain areas of code that are both simple and cannot fail. Revocation logic is one. Authentication logic is another. These are the core components of a wallet. One can imagine extending a wallet further with plugins in way were failure is more tolerable. Suppose a user adds a handful of plugins which bit rot over time. Revoking that plugin should be simple and possible via the core wallet implementation.
+
+It's important to note that revocation is only useful as a mitigation against bugs that have not yet been found. As soon as a vulnerability exists in a wallet, the funds in the account must be considered as MEV and value of the account will be auctioned off.
+
+This is not to say that revocation isn't critical. It is simply that smart contracts and EOA wallets are capable of providing the same functionality. Even if the there is no cost to the user for in-protocol revocation, it is undue complexity in the core protocol. We must resist all except which is absolutely necessary to encode in the core protocol.
+
+### Setting code as `tx.origin`
+
+Allowing `tx.origin` to set code enables simple transaction batching, where the sender of the outer transaction would be the signing account. The ERC-20 approve-then-transfer pattern, which currently requires two separate transactions, could be completed in a single transaction with this proposal.
+
+`AUTH` allows for signatures to be signed by `tx.origin`. For any such signatures, subsequent `AUTHCALL`s have `msg.sender == tx.origin` in their first layer of execution. Without EIP-7702, this situation can only ever arise in the topmost execution layer of a transaction. This EIP breaks that invariant and so affects smart contracts containing `require(msg.sender == tx.origin)` checks. This check can be used for at least three purposes:
```

Comment:
Why are we mentioning AUTH and AUTHCALL here?

Reply:
i lifted some of this rationale from 3074, i fixed this.